### DISCUSSION TITLE
Root level task execution

### DISCUSSION THREAD

**Discussion by sppatel** [vercel/turborepo#1057] [Help] [answered]
As part of turborepo adoption, what is the general guidance on tasks that were previously run as a root level task because its more efficient to do so (vs individual packages).  For example, currently we run tests (jest) at the root of the repo - this is much faster because it avoids jest having to duplicate file traversal across the same files which is extremely expensive.   

I am assuming that turborepo is expecting all tasks to be run as package operations - which is potentially a tradeoff (leverage turborepo task running at the cost of inefficiency of running jest processes multiple times).   It is not clear to me what approach comes out on top.

I am assuming others have dealt with the jest use case in a mono repo - and would love to hear experiences moving from a root level to per package operation.

Thank you.

**Comment by sppatel** [comment, marked as answer]
After actually adopting turbo, I think the answer is to switch from root level execution to per package.  The net benefit is greater then running under a single process.

**Reply by lodmfjord** [reply]
This is the same conclusion I have come to. This can make initial process little bit slower when nothing is cached but the benefits when something or nearly everything is cached is very great.

**Reply by nareshbhatia** [reply]
@sppatel do you have a sample repo showing your jest approach?

P.S. I am looking at the kitchen sink example and not sure if I understand it fully. For one thing, if doesn't seem to be set `testEnvironment: 'jsdom'` anywhere.
